src/aws-cpp-sdk-core/source/http/crt/CRTHttpClient.cpp

```diff
@@ -246,13 +246,19 @@ namespace Aws
             }
 
             //TODO: handle the read rate limiter here, once back pressure is setup.
+            assert(response);
             for (const auto& hashIterator : request->GetResponseValidationHashes())
             {
+              std::stringstream headerStr;
+              headerStr<<"x-amz-checksum-"<<hashIterator.first;
```

Comment:
An improvement here would be to embed the c string representation of the header into the hash class, avoiding a stringstream to string to c string for every buffer that arrives.